src/main/java/com/google/devtools/build/lib/remote/blobstore/http/HttpBlobStore.java

```diff
@@ -267,6 +268,8 @@ private Channel acquireUploadChannel() throws InterruptedException {
                 p.addLast(new HttpObjectAggregator(10 * 1024));
                 p.addLast(new HttpRequestEncoder());
                 p.addLast(new ChunkedWriteHandler());
+                p.addFirst("read-timeout-handler", new ReadTimeoutHandler(timeoutSeconds));
+                p.addFirst("write-timeout-handler", new WriteTimeoutHandler(timeoutSeconds));
```

Comment:
The semantics of `WriteTimeoutHandler` are such that it will trigger a timeout if a write doesn't finish in `timeoutSeconds`. Now, for HttpUploaderHandler one file is one write (because we use `HttpChunkedInput`) and this file is only chunked up into smaller bytes slices in `ChunkedWriteHandler`. That's why the WriteTimeoutHandler at the very least needs to be added before the `ChunkedWriteHandler` but I believe it makes to add it to the very beginning as we also wan to have a timeout on any additional writes being done by netty's http stack.

Reply:
I will change the order of the handlers and see what happens!

Reply:
I changed the code to use an idle timeout handler.  Either reads or writes should reset the timeout countdown clock.